### PR TITLE
Bugfix: etcd tests

### PR DESCRIPTION
**Pull request by aolfat**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: metadata/etcd_test.go [hunk 1]
```diff
@@ -689,46 +690,76 @@ func TestEtcdConfig_Put(t *testing.T) {
...
+	if err == nil {
+		t.Errorf("Get() should have failed with invalid key")
+	}
+	// check to see if it's an etcderror
+	if _, ok := err.(*KeyNotFoundError); !ok {
+		t.Errorf("Get() should have failed with KeyNotFoundError")
+	}
+}
+
+func TestEtcdConfig_ValidKey(t *testing.T) {
```
Comment (ahmadnazeri) [inline metadata/etcd_test.go:744]: LGTM but we can change the tests into test cases instead of separate tests. For now, we are good.

**Review by ahmadnazeri — APPROVED** [review]

File: metadata/etcd_test.go (hunk 1 above)
Comment (aolfat) [reply metadata/etcd_test.go:744]: why split it up?

Comment (aolfat) [reply metadata/etcd_test.go:744]: why do test cases?

Comment (aolfat) [reply metadata/etcd_test.go:744]: I actually tried to it at first. They're slightly different enough where it actually makes the one bigger test harder to read.

Comment (aolfat) [reply metadata/etcd_test.go:744]: let me try one more time

Comment (aolfat) [reply metadata/etcd_test.go]: alright i think i made it better

**Review by sdreyer — APPROVED** [review]

**Pull request merged**